cpp/benchmarks/common/random_distribution_factory.hpp

```diff
@@ -18,17 +18,29 @@
 
 #include "generate_input.hpp"
 
-#include <memory>
-#include <random>
+#include <rmm/device_uvector.hpp>
+
+#include <thrust/execution_policy.h>
+#include <thrust/random.h>
+#include <thrust/random/normal_distribution.h>
+#include <thrust/random/uniform_int_distribution.h>
+#include <thrust/tabulate.h>
 
+#include <cmath>
+#include <memory>
+#include <type_traits>
 /**
  * @brief Generates a normal(binomial) distribution between zero and upper_bound.
  */
 template <typename T, std::enable_if_t<cuda::std::is_integral_v<T>, T>* = nullptr>
 auto make_normal_dist(T upper_bound)
 {
-  using uT = typename std::make_unsigned<T>::type;
-  return std::binomial_distribution<uT>(upper_bound, 0.5);
+  // Provided n is large enough, Normal(μ,σ2) is a good approximation for Binomial(n, p)
+  // where μ = np and σ2 = np (1 - p).
```

Comment:
While true, this statement is confusing because it's not clear why we want to approximate a binomial distribution. What is the underlying data distribution that we are trying to mimic? What guarantees do we have on the size of the data? Do we care how good this approximation is?

Reply:
`std::normal_distribution<RealType>` The effect is undefined if this is not one of float, double, or long double.
https://stackoverflow.com/a/22016408
binomial is an approximation of normal distribution for integral types.

Actually, we don't need binomial approximation here since thrust does not have binomial. We need to use normal distribution with `std::round` for integral types.
I will update the code accordingly.

Reply:
might have to check if the floating point value is within integer range, I don't think `round` behaves well in such cases.

Reply:
Same request as [above](https://github.com/rapidsai/cudf/pull/10109#discussion_r828466145) regarding docs.